sebsync.py

```diff
@@ -138,42 +151,84 @@ def get_remote_ebooks() -> None:
         click.echo(f"Found {len(remote_ebooks)} remote ebooks.")
 
 
-def get_local_ebooks() -> None:
+def get_local_ebooks(local_cache: dict) -> None:
     """Retrieve metadata of Standard EPUBs in the specified directory and subdirectories."""
-    for path in options.books.glob("**/*.epub"):
-        if not path.is_file():
-            continue
-        try:
-            with zipfile.ZipFile(path) as zip:
-                with zip.open("META-INF/container.xml") as file:
-                    root = ElementTree.parse(file)
-                    ns = {"container": "urn:oasis:names:tc:opendocument:xmlns:container"}
-                    rootfile = root.find(".//container:rootfile", ns).attrib["full-path"]
-                with zip.open(rootfile) as file:
-                    root = ElementTree.parse(file)
-                    ns = {
-                        "opf": "http://www.idpf.org/2007/opf",
-                        "dc": "http://purl.org/dc/elements/1.1/",
-                    }
-                    metadata = root.find("opf:metadata", ns)
-                    id = metadata.find("dc:identifier", ns)
-                    if id is None or "standardebooks.org" not in id.text:
-                        continue
-                    modified = metadata.find(".//opf:meta[@property='dcterms:modified']", ns)
+    if options.type == "kindle":
+        # Retrieve the id, title, path, modified date, and hexdigest from our cache file
+        for path in options.books.glob("**/*.azw3"):
+            if not path.is_file():
+                continue
+            try:
+                if path.name in local_cache:
```

Comment:
If the file were moved or renamed, how would it be located in the index? (See my comment below on line 171.) I suggest that the hash itself become the dictionary key (or alternatively, if you prefer a more structured approach, just a list of dicts with key-value pairs for each book, `hexdigest` being one of the keys as you're doing below.)